### PR TITLE
NXP-32228: fix Legal hold warning popup should not appear for undeclared record

### PR DESCRIPTION
**Pull request by swarnadipa-dev**
https://jira.nuxeo.com/browse/NXP-32228

### CODE REVIEW

File: nuxeo-retention-web/elements/nuxeo-hold-toggle-button.js [hunk 1]
```diff
@@ -145,8 +145,11 @@ class RetentionHoldToggleButton extends mixinBehaviors([FiltersBehavior, FormatB
   }
 
   _hold() {
-    const { isFlexibleRecord } = this.document;
-    const response = isFlexibleRecord ? window.confirm(this.i18n('retention.holdToggleButton.confirm.hold')) : true;
+    const { isFlexibleRecord, isUnderRetentionOrLegalHold } = this.document;
+    const response =
+      isFlexibleRecord && isUnderRetentionOrLegalHold
```
Comment (guirenard) [inline nuxeo-retention-web/elements/nuxeo-hold-toggle-button.js:150]: Why do we need to check in addition `isUnderRetentionOrLegalHold` before warning?

Comment (guirenard) [reply nuxeo-retention-web/elements/nuxeo-hold-toggle-button.js:150]: maybe you meant `isFlexibleRecord && isRecord` ?

Comment (Nishant0928) [reply nuxeo-retention-web/elements/nuxeo-hold-toggle-button.js:150]: when we unattached the rule on the document , `isUnderRetentionOrLegalHold` is the only property which changes from true to false , so in one scenario where a user unattach a rule and then apply legal hold , the pop up confirmation should not appear , so `isUnderRetentionOrLegalHold` is the only property which helps to achieve this.

Comment (guirenard) [reply nuxeo-retention-web/elements/nuxeo-hold-toggle-button.js:150]: so be it

**Review by rahuljain-dev — APPROVED** [review]

**Review by Nishant0928 — APPROVED** [review]

**Review by arupmmi07 — APPROVED** [review]
ok

**Review by guirenard — APPROVED** [review]

**Pull request merged**